pyccel/parser/semantic.py

```diff
@@ -902,11 +914,11 @@ def _handle_kernel(self, expr, func, args, **settings):
                         symbol = expr,
                         severity='fatal')
             # TODO : type check the NUMBER OF BLOCKS 'numBlocks' and threads per block 'tpblock'
-            if not isinstance(expr.numBlocks, LiteralInteger):
+            if not isinstance(expr.numBlocks, (LiteralInteger, PyccelSymbol)):
```

Comment:
we either check exactly on if the expr is a int `variable` if we can or we just limit it to `LiteralInteger` for now

Reply:
For this one i added PyccelSymbol because it seems like you are variables as number or blocks and number of thread per blocks in other cuda test.